File: linked_list/problem_solutions/intersection_of_two_linked_lists.c
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode *getIntersectionNode(struct ListNode *headA, struct ListNode *headB) {
    struct ListNode *pA = headA, *pB = headB;
    while (pA != pB)
    {
        pA = pA ? pA->next : headB;
        pB = pB ? pB->next : headA;
    }
    return (pA);
}

File: linked_list/problem_solutions/intersection_of_two_linked_lists.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
    pointerA = headA
    pointerB = headB

    while (pointerA != pointerB):
        pointerA = pointerA.next if pointerA != None else headB
        pointerB = pointerB.next if pointerB != None else headA
    
    return pointerA